### PR TITLE
Expose HttpHandler Decoration as a bean

### PR DESCRIPTION
**Pull request by sp00m**
Follows the development that has been done for https://github.com/spring-projects/spring-framework/issues/25633.

Having the `httpHandlerDecorator` as a bean allows easier integration with Spring Boot, see [`HttpHandlerAutoConfiguration`](https://github.com/spring-projects/spring-boot/blob/v2.4.2/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/reactive/HttpHandlerAutoConfiguration.java#L65).

I wasn't sure what version to put in `@since` Javadoc tags.

Let me know what you think.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@sp00m Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=26502)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-framework&pullRequestId=26502) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@sp00m Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=26502)!

**Comment by rstoyanchev** [comment]
Something like this would be fine.

**Comment by rstoyanchev** [comment]
A couple of things I'll probably adjust. 1) we'd want to allow multiples of these, and 2) it should be named something other than Decorator for consistency with other such types in our web code which normally are implementations of the target type (`HttpHandlerDecoratorFactory` perhaps).

**Comment by sp00m** [comment]
@rstoyanchev, thanks for the review, updated as proposed, let me know.

File: spring-web/src/test/java/org/springframework/web/server/adapter/WebHttpHandlerBuilderTests.java [hunk 1]
```diff
@@ -33,13 +25,22 @@
...
 import org.springframework.web.testfixture.http.server.reactive.MockServerHttpRequest;
 import org.springframework.web.testfixture.http.server.reactive.MockServerHttpResponse;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.BiFunction;
```
Comment (sp00m) [inline spring-web/src/test/java/org/springframework/web/server/adapter/WebHttpHandlerBuilderTests.java:43]: Oops, just noticed those modifications. IntelliJ reordered them on my behalf, let me know if you want me to revert them to reduce the diff. I use IntelliJ's default config FYI.

**Pull request closed without merging**